readthedocs/oauth/models.py

```diff
@@ -59,17 +50,54 @@ class RemoteOrganization(models.Model):
         null=True,
         blank=True,
     )
-
-    json = models.TextField(_('Serialized API response'))
+    remote_id = models.CharField(
+        db_index=True,
+        max_length=128
+    )
+    vcs_provider = models.CharField(
+        _('VCS provider'),
+        choices=VCS_PROVIDER_CHOICES,
+        max_length=32
+    )
 
     objects = RemoteOrganizationQuerySet.as_manager()
 
+    class Meta:
+        ordering = ['name']
+        unique_together = (('remote_id', 'vcs_provider'),)
+        db_table = 'oauth_remoteorganization_2020'
+
     def __str__(self):
         return 'Remote organization: {name}'.format(name=self.slug)
 
+
+class RemoteOrganizationRelation(TimeStampedModel):
+    remote_organization = models.ForeignKey(
+        RemoteOrganization,
+        related_name='remote_organization_relations',
+        on_delete=models.CASCADE
+    )
+    user = models.ForeignKey(
+        User,
+        related_name='remote_organization_relations',
+        on_delete=models.CASCADE
+    )
+    account = models.ForeignKey(
+        SocialAccount,
+        verbose_name=_('Connected account'),
+        related_name='remote_organization_relations',
+        null=True,
+        blank=True,
```

Comment:
What would be the case where we do have a relation with user, but the account is null? I think this is not possible.

Reply:
Yes you are right, maybe we did it because of the previous plan